# This change is Solaris-specific and thus is not being contributed back
# to the upstream community.  Details:
#
# OpenSSH updates lastlog when users login however on Solaris the
# pam_open_session(3PAM) function handles this so we avoid the
# redundant additional upating of lastlog here.  We can't disable all
# lastlog references with --disable-lastlog or NO_SSH_LASTLOG since that
# prevents the 'Last login: ...' message from being displayed.  Thus
# we just prevent OpenSSH from updating lastlog since reading from
# lastlog is fine.  Other PAM implementations don't update lastlog
# during session management so this change is Solaris-specific and
# thus isn't being contributed back to the upstream community.
#
*** orig/loginrec.c	Thu Oct  4 16:08:42 2012
--- new/loginrec.c      Thu Sep  4 16:41:24 2014
***************
*** 444,450 ****
  #ifdef USE_LOGIN
  	syslogin_write_entry(li);
  #endif
! #ifdef USE_LASTLOG
  	if (li->type == LTYPE_LOGIN)
  		lastlog_write_entry(li);
  #endif
--- 444,450 ----
  #ifdef USE_LOGIN
  	syslogin_write_entry(li);
  #endif
! #if defined(USE_LASTLOG) && !defined(PAM_SUN_CODEBASE)
  	if (li->type == LTYPE_LOGIN)
  		lastlog_write_entry(li);
  #endif
